packages/test-utils/src/find-vnodes.js

```diff
@@ -27,49 +18,23 @@ function removeDuplicateNodes (vNodes: Array<VNode>): Array<VNode> {
   return vNodes.filter((vNode, index) => index === vNodes.findIndex(node => vNode.elm === node.elm))
 }
 
-function nodeMatchesRef (node: VNode, refName: string): boolean {
-  return node.data && node.data.ref === refName
-}
-
-function findVNodesByRef (vNode: VNode, refName: string): Array<VNode> {
+export function findVNodesByRef (
+  vNode: VNode,
+  refName: string
+): Array<VNode> {
   const nodes = findAllVNodes(vNode)
-  const refFilteredNodes = nodes.filter(node => nodeMatchesRef(node, refName))
-  // Only return refs defined on top-level VNode to provide the same
-  // behavior as selecting via vm.$ref.{someRefName}
-  const mainVNodeFilteredNodes = refFilteredNodes.filter(node => (
-    !!vNode.context.$refs[node.data.ref]
-  ))
-  return removeDuplicateNodes(mainVNodeFilteredNodes)
+    .filter(node => node.data && node.data.ref === refName && !!vNode.context.$refs[node.data.ref])
+  return removeDuplicateNodes(nodes)
 }
 
-function nodeMatchesSelector (node: VNode, selector: string): boolean {
-  return node.elm && node.elm.getAttribute && node.elm.matches(selector)
+export function elementMatchesSelector (element: any, selector: string) {
+  return !!(element && element.getAttribute && element.matches(selector))
 }
 
-function findVNodesBySelector (
+export function findVNodesByDOMSelector (
   vNode: VNode,
   selector: string
 ): Array<VNode> {
-  const nodes = findAllVNodes(vNode)
-  const filteredNodes = nodes.filter(node => (
-    nodeMatchesSelector(node, selector)
-  ))
-  return removeDuplicateNodes(filteredNodes)
-}
-
-export default function findVnodes (
```

Comment:
This method is remove in favor of `findVNodesByDOMSelector` and `findVNodesByRef`. Checking for `$ref selectors can only be used on Vue component wrappers` and distinction of `selectorType` is moved up into `find.js`